File: QuickSort.java
import java.util.Arrays;

public class QuickSort {
	
	public static void main(String[] args) {
		
		int[] arr= {15,9,7,13,12,16,4,18,11};
		int leng =arr.length;
		QuickSort qsm=new QuickSort();
		qsm.quicksortRecursion(arr,0,leng-1);
		System.out.println(Arrays.toString(arr));
		
	}
	
	int partition(int []arr,int low,int high)
	{
		int pivot=arr[(low+high)/2];
		
		while(low<=high)
		{
			while(arr[low]<pivot)
			low++;
			
			while(arr[high]>pivot)
			high--;
			
			if(low<=high)
			{
				int temp=arr[low];
				arr[low]=arr[high];
				arr[high]=temp;
				low++;high--;
			}
		}
		return low;
	}
	void quicksortRecursion(int[] arr,int low,int high)
	{
		int pi=partition(arr,low,high);
		if(low<pi-1)
		quicksortRecursion(arr,low,pi-1);
		
		if(pi<high)
		quicksortRecursion(arr,pi,high);
		
	}

}

File: Quick Sort.cpp
#include<bits/stdc++.h>
using namespace std;
void swap(int arr[],int i,int j)
    {
        int temp=arr[i];
        arr[i]=arr[j];
        arr[j]=temp;
    }
int partition(int arr[],int l,int r)
    {
        int pivot=arr[r];
        int i=l-1;
        for(int j=l;j<r;j++)
        {
            if(arr[j]<pivot)
            {
                i++;
                swap(arr,i,j);
            }
        }
        swap(arr,i+1,r);
        return i+1;
    }
void quickSort(int arr[],int l,int r)
    {
        if(l<r)
        {
        	int pi=partition(arr,l,r);
        	
        	quickSort(arr,l,pi-1);
        	quickSort(arr,pi+1,r);
		}
    }
int main()
{
	int n;
	cin>>n;
	int arr[n];
	for(int i=0;i<n;i++)
	{
		cin>>arr[i];
	}
	quickSort(arr,0,n-1);
	for(int i=0;i<n;i++)
	{
		cout<<arr[i]<<" ";
	}
	return 0;
}    
